types/src/account_state_blob/mod.rs

```diff
@@ -89,26 +76,42 @@ impl From<AccountStateBlob> for crate::proto::types::AccountStateBlob {
     }
 }
 
-#[cfg(any(test, feature = "fuzzing"))]
-impl From<AccountResource> for AccountStateBlob {
-    fn from(account_resource: AccountResource) -> Self {
-        let mut account_state: BTreeMap<Vec<u8>, Vec<u8>> = BTreeMap::new();
-        account_state.insert(
-            ACCOUNT_RESOURCE_PATH.to_vec(),
-            lcs::to_bytes(&account_resource).unwrap(),
-        );
-        AccountStateBlob::try_from(&account_state).unwrap()
+impl TryFrom<&AccountState> for AccountStateBlob {
+    type Error = Error;
+
+    fn try_from(account_state: &AccountState) -> Result<Self> {
+        Ok(Self {
+            blob: lcs::to_bytes(account_state)?,
+        })
     }
 }
 
-impl TryFrom<&AccountStateBlob> for BTreeMap<Vec<u8>, Vec<u8>> {
+impl TryFrom<&AccountStateBlob> for AccountState {
     type Error = Error;
 
     fn try_from(account_state_blob: &AccountStateBlob) -> Result<Self> {
         lcs::from_bytes(&account_state_blob.blob).map_err(Into::into)
     }
 }
 
+impl TryFrom<&AccountResource> for AccountStateBlob {
```

Comment:
Same here. If account resource and account state blob is not a 1:1 mapping. I feel this conversion is a bit weird.